File: balancedBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */

var isBalanced = function(root) {
    if(root == null) return true;
    let leftHeight = helper(root.left,0);
    let rightHeight = helper(root.right,0);
    if(Math.abs(leftHeight - rightHeight)>1) return false;
    return isBalanced(root.left) && isBalanced(root.right);
     
};

var helper = function(root,count){
    //base
    if(root == null) return count;
    //logic
    count = count+1;
    let leftcount = helper(root.left,count);
    let rightcount = helper(root.right,count);
    return Math.max(leftcount,rightcount);
    
};

File: balancedBinaryTree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

package main

func isBalanced(root *TreeNode) bool {
	if root == nil {
		return true
	}
	leftHeight := maxHeight(root.Left)
	rightHeight := maxHeight(root.Right)
	if leftHeight > rightHeight {
		if leftHeight-rightHeight > 1 {
			return false
		}
	} else {
		if rightHeight-leftHeight > 1 {
			return false
		}
	}
	return isBalanced(root.Left) && isBalanced(root.Right)
}

func maxHeight(root *TreeNode) int {
	if root == nil {
		return 0
	}
	left := maxHeight(root.Left)
	right := maxHeight(root.Right)
	if left > right {
		return left + 1
	} else {
		return right + 1
	}
}
